TypeError: expected str, bytes or os.PathLike object, not NoneType in CaC/content

This seems to be the same error message from #23

`python3 ./ctf/content_test_filtering.py pr --rule --output json 7908 > output.json`

```
Traceback (most recent call last):
  File "/__w/content/content/./ctf/content_test_filtering.py", line 44, in <module>
    diff_structure = diff_analysis.analyse_file(file_record)
  File "/__w/content/content/ctf/ctf/diff_analysis.py", line 54, in analyse_file
    return file_analyzer.process_analysis()
  File "/__w/content/content/ctf/ctf/analysis/BashAnalysis.py", line 103, in process_analysis
    self.diff_struct.add_changed_product_by_rule(self.rule_name, msg=msg)
  File "/__w/content/content/ctf/ctf/DiffStruct.py", line 122, in add_changed_product_by_rule
    product_name = self.get_rule_products(rule_name)
  File "/__w/content/content/ctf/ctf/DiffStruct.py", line 89, in get_rule_products
    with open(ruleyml_path) as f:
TypeError: expected str, bytes or os.PathLike object, not NoneType
Error: Process completed with exit code 1.
```

--- a/ctf/analysis/AnsibleAnalysis.py
+++ b/ctf/analysis/AnsibleAnalysis.py
@@ -11,8 +11,12 @@ class AnsibleAnalysis(AbstractAnalysis):
     def __init__(self, file_record):
         super().__init__(file_record)
         self.diff_struct.file_type = FileType.YAML
-        self.rule_name = re.match(r".+/((?:\w|-)+)/ansible/\w+\.yml$",
-                                  self.filepath).group(1)
+        rule_name_match = re.match(r".+/((?:\w|-)+)/ansible/((?:\w|-)+)\.yml$",
+                                   self.filepath)
+        if rule_name_match.group(1) == "fixes":
+            self.rule_name = rule_name_match.group(2)
+        else:
+            self.rule_name = rule_name_match.group(1)
 
     @staticmethod
     def can_analyse(filepath):

--- a/ctf/analysis/BashAnalysis.py
+++ b/ctf/analysis/BashAnalysis.py
@@ -14,10 +14,10 @@ class BashAnalysis(AbstractAnalysis):
         self.diff_struct.file_type = FileType.BASH
         rule_name_match = re.match(r".+/((?:\w|-)+)/bash/((?:\w|-)+)\.sh$",
                                    self.filepath)
-        if rule_name_match.group(2) == "shared":
-            self.rule_name = rule_name_match.group(1)
-        else:
+        if rule_name_match.group(1) == "fixes":
             self.rule_name = rule_name_match.group(2)
+        else:
+            self.rule_name = rule_name_match.group(1)
 
     @staticmethod
     def can_analyse(filepath):

--- a/ctf/analysis/OVALAnalysis.py
+++ b/ctf/analysis/OVALAnalysis.py
@@ -17,8 +17,12 @@ class OVALAnalysis(AbstractAnalysis):
     def __init__(self, file_record):
         super().__init__(file_record)
         self.diff_struct.file_type = FileType.OVAL
-        self.rule_name = re.match(r".+/((?:\w|-)+)/oval/\w+\.xml$",
-                                  self.diff_struct.absolute_path).group(1)
+        rule_name_match = re.match(r".+/((?:\w|-)+)/oval/((?:\w|-)+)\.xml$",
+                                   self.filepath)
+        if rule_name_match.group(1) == "checks":
+            self.rule_name = rule_name_match.group(2)
+        else:
+            self.rule_name = rule_name_match.group(1)
         self.tree_before = None
         self.tree_after = None
 

